[Merged by Bors] - Add additional libp2p tests

pawanjay176 opened the pull request:
## Issue Addressed

N/A

## Proposed Changes

Adds tests for the eth2_libp2p crate.

pawanjay176 commented:
Added some unit tests for rpc codecs and some of the types in `eth2_libp2p`.

@AgeManning @divagant-martian @blacktemplar Please let me know if there are any other parts of `eth2_libp2p` that you feel needs to have more tests.

AgeManning commented on beacon_node/eth2_libp2p/src/types/subnet.rs:
```diff
@@ -21,8 +21,52 @@ impl PartialEq for SubnetDiscovery {
                 (None, None) => true,
-                (None, Some(_)) => true,
-                (Some(_), None) => true,
+                (None, Some(_)) => false, // not equal
```
Can you confirm you're not breaking attestation subnet logic here. 

This was implemented specifically for a reason. I recall this logic being somewhat critical in deciding if we need to search for new subnets or not. 

I agree the current form is not strictly correct from a `PartialEq` standpoint but perhaps its correct for its use in the attestation service

pawanjay176 replied:
> This was implemented specifically for a reason. I recall this logic being somewhat critical in deciding if we need to search for new subnets or not.

Thanks for pointing it out. I think this was needed before the attestation service refactor. Removing the `PartialEq` impl entirely only affected the tests att service tests. 

I'll run this on medalla for a bit and double check the logic to make sure.

pawanjay176 replied:
Okay, I'm convinced that we aren't using the `PartialEq` impl anywhere else but in the attestation service tests. 
For e.g. https://github.com/sigp/lighthouse/blob/master/beacon_node/network/src/attestation_service/tests/mod.rs#L208

Also ran this on medalla for about 2 hours with 0 missed attestations.
So I'm pretty sure this doesn't break any existing logic.

However, since we don't compare any 2 `SubnetDiscovery` structs directly, maybe we can remove the `PartialEq` impl entirely or have it behind a `#cfg(test)`  flag?

AgeManning replied:
Yeah, lets gate it behind the #cfg(test). 

Thanks :)

AgeManning commented:
I noticed you removed the option and kept the partial eq? 

I thought the partial eq wasn't necessary for anything? Also IIRC the option was for when we wanted to subscribe to long-lived subnets. Perhaps we don't need it anymore?

pawanjay176 commented:
@AgeManning moved the `SubnetDiscovery` `PartialEq` impl to attestation service as `#[cfg(test)]` across crates(eth2_libp2p and network in this case) does not work. Fixed the snappy test and added some discovery tests.

AgeManning reviewed (approved):
Nice work! LGTM

AgeManning commented:
bors r+
